pkg/reconciler/revision/reconcile_resources.go

```diff
@@ -116,9 +116,13 @@ func (c *Reconciler) reconcileDeployment(ctx context.Context, rev *v1.Revision)
 						} else {
 							rev.Status.MarkContainerHealthyFalse(v1.ExitCodeReason(t.ExitCode), v1.RevisionContainerExitingMessage(t.Message))
 						}
-					} else if w := status.State.Waiting; w != nil && hasDeploymentTimedOut(deployment) {
-						logger.Infof("marking resources unavailable with: %s: %s", w.Reason, w.Message)
-						rev.Status.MarkResourcesAvailableFalse(w.Reason, w.Message)
+					} else if w := status.State.Waiting; w != nil {
+						if hasDeploymentTimedOut(deployment) {
+							logger.Infof("marking resources unavailable with: %s: %s", w.Reason, w.Message)
+							rev.Status.MarkResourcesAvailableFalse(w.Reason, w.Message)
+						} else {
+							rev.Status.PropagateDeploymentAvailabilityStatusIfFalse(&deployment.Status)
```

Comment:
Without this we can't report failures outside a rolout see https://github.com/kubernetes/kubernetes/issues/106054 for more.